planex/makesrpm.py

```diff
@@ -0,0 +1,133 @@
+import subprocess
+import os
+from shutil import copyfile
+import fileinput
+import tarfile
+import tempfile
+import shutil
+
+import argparse
+import argcomplete
+from planex.util import add_common_parser_options
+
+
+def parse_args_or_exit(argv):
+    """
+    Parse command line options
+    """
+    parser = argparse.ArgumentParser(description='Planex SRPM building')
+    add_common_parser_options(parser)
+    parser.add_argument(
+        "--topdir", metavar="DIR", default=None,
+        help='Set rpmbuild toplevel directory')
+    parser.add_argument(
+        "--dist", metavar="DIST", default="",
+        help="distribution tag (used in RPM filenames)")
+    argcomplete.autocomplete(parser)
+    return parser.parse_known_args(argv)
+
+
+def parse_patchseries(series_file):
+    """
+    Parse series file and return the list of patches
+    """
+    patches = []
+    for line in open(series_file):
+        li = line.strip()
+        if not li.startswith("#"):
+            patches.append(li)
+
+    return patches
+
+
+def main(argv):
+    """
+    Entry point - Arguments: component.spec Source0 Source1 ... Patch0 Patch1 ...
+    The specfile must always be present and it has to be the first in the list
+    """
+    intercepted_args, passthrough_args = parse_args_or_exit(argv)
+    target = os.path.basename(passthrough_args[0]).split('.')[0]
+    specfile = passthrough_args[0]
+    tmp_dirpath = tempfile.mkdtemp() + '/'
+    tmp_specs = tmp_dirpath + 'SPECS/'
+    tmp_specfile = tmp_dirpath + specfile
+    tmp_build = tmp_dirpath + '_build/'
+    tmp_sources = tmp_build + 'SOURCES/'
+
+    # Create temporary working area
+    os.makedirs(tmp_specs)
+    os.makedirs(tmp_build)
+    os.makedirs(tmp_sources)
+
+    try:
+        # Copy files to temporary working area
+        copyfile(specfile, tmp_specfile)
+        patchqueue = False
+        patchqueue_path = None
+        patchqueue_filters = ['.pg.', '.pq.']
+
+        for source in passthrough_args[1:]:
+            copyfile(source, tmp_dirpath + source)
+            if any(map(lambda x: x in source, patchqueue_filters)):
+                patchqueue = True
+                patchqueue_path = source
+
+        if patchqueue:
+            # Extract patches
+            patch_number = 0
+            for line in fileinput.input(tmp_specfile, inplace=True):
+                if any(map(lambda x: x in line, patchqueue_filters)):
+                    tar = tarfile.open(tmp_dirpath + patchqueue_path)
+                    for f in tar.getmembers():
+                        f.name = '%s-' % target + os.path.basename(f.name)
+                        tar.extract(f, tmp_sources)
+                    patches = parse_patchseries(tmp_sources +
+                                                '%s-series' % target)
+                    print "# Patches for %s" % target
+                    for p in patches:
+                        if p != "":
+                            print "Patch%s: " % patch_number +
```

Comment:
You can do this in one string interpolation step:
`print "Patch%s: %%{name}-%s" % (patch_number, p)`